source/develop/release-management/features/network/networking-api-security-groups.html.md

```diff
@@ -0,0 +1,468 @@
+default - this means an option for using oVirt's external networks
+without security group support should be possible.
+
+* If the external network has a subnet defined, DHCP traffic is allowed to
+/ from the VM.
+
+### Requirements
+
+* For ingress traffic (to an instance)
+  + Only traffic matched with security group rules are allowed.
+  + When there is no rule defined, all traffic are dropped.
+* For egress traffic (from an instance)
+  + Only traffic matched with security group rules are allowed.
+  + When there is no rule defined, all egress traffic are dropped.
+  + When a new security group is created, rules to allow all egress traffic are automatically added.
+
+### Security groups description
+
+Security groups, in the general sense, are nothing more than containers
+for [security group rules](https://developer.openstack.org/api-ref/network/v2/#security-group-rules-security-group-rules). 
+They are applied to logical ports, and thus are used to associate a set of 
+rules to a port.
+
+They are are indeed a white list - e.g. everything is *denied* by
+default, and the user adds rules that allow traffic. As long as *one*
+rule is matched, the traffic is allowed.
+
+Security group rules can match on L3 and L4 parameters, specifically on
+the protocol (L3), the IP protocol (L4), and destination ports. The remote IP
+address is also a subject for the match conditions, and rules can be
+added to allow *incoming* and *outgoing* traffic.
+
+The API of security group rules can be found [here](https://developer.openstack.org/api-ref/network/v2/#security-group-rules-security-group-rules).
+
+### Activating the feature
+
+Activating the feature means one thing: have the ***deny-all*** security group
+provisioned. 
+
+To do so, two alternatives are possible, and both will result in the group 
+being provisioned:
+
+1. Add a configuration flag to the ovirt-provider-ovn configuration file. When 
+the ovirt-provider-ovn service is started, if this property is configured, then
+the **deny-all** security group will be provisioned - if not alredy.
+
+2. In the engine-setup, present a new question to the oVirt administrator, 
+asking if he wants the security group feature activated. An affirmative reply
+will provision the **deny-all** security group.
+
+### User perspective
+
+The user is meant to provision security group(s) and rule(s) through ansible,
+or directly using a REST client.
+
+This greatly reduces the oVirt-engine impact, since there's no need to 
+implement the REST client side of the API, nor implement these changes in the
+GUI.
+
+Throughout the remainder of the document, ansible will be assumed as the client, given its descriptive yaml syntax. The [os_security_group](https://docs.ansible.com/ansible/2.5/modules/os_security_group_module.html) and [os_security_group_rules](https://docs.ansible.com/ansible/2.5/modules/os_security_group_rule_module.html) will be leveraged for this.
+
+### Mandatory group creation
```

Comment:
i'm considering adding plantuml diagrams for this flow. 

do you think the text alone is easy to understand?